### PR TITLE
add info to README

### PR DESCRIPTION
**Pull request by heyitsvi**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,110 @@
 # cmake-pear
+This repository contains essential CMake functions and modules to streamline the build process and packaging of Pear applications within the [bare-dev](https://github.com/holepunchto/bare-dev) development environment.
 
 ## API
+`add_pear_appling function`: Streamlines the process of defining and configuring  a new Pear appling. Takes care of:
+- Linking the code to the core Pear libraries.
+- Managing appling metadata (key, name, version, etc.).
+- Calling the appropriate platform-specific packaging functions.
```
Comment (kasperisager) [inline README.md]:
```suggestion

#### `add_pear_appling`

Defines and configures a new Pear appling. Takes care of:

- Linking the code to the core Pear libraries.
- Managing appling metadata (key, name, version, etc.).
- Calling the appropriate platform-specific packaging functions.
```

File: README.md [hunk 2]
```diff
@@ -1,8 +1,110 @@
 # cmake-pear
+This repository contains essential CMake functions and modules to streamline the build process and packaging of Pear applications within the [bare-dev](https://github.com/holepunchto/bare-dev) development environment.
```
Comment (kasperisager) [inline README.md]:
```suggestion

CMake functions to streamline the build process and packaging of Pear applications.
```

File: README.md [hunk 3]
```diff
@@ -1,8 +1,110 @@
...
+MACOS_IDENTIFIER <string>
+MACOS_SIGNING_IDENTITY <string>
+[MACOS_SIGNING_KEYCHAIN <string>]
+[MACOS_ENTITLEMENTS <entitlement...>]
+[WINDOWS_ICON <path>]
+WINDOWS_SIGNING_SUBJECT <string>
+WINDOWS_SIGNING_THUMBPRINT <string>
+[LINUX_ICON <path>]
+LINUX_CATEGORY <string>)
+```
```
Comment (kasperisager) [inline README.md]:
```suggestion
```cmake
add_pear_appling(
  <target>
  KEY <string> 
  NAME <string> 
  VERSION <string> 
  DESCRIPTION <string> 
  AUTHOR <string> 
  [SPLASH <path>]
  [MACOS_ICON <path>]
  MACOS_CATEGORY <string>
  MACOS_IDENTIFIER <string>
  MACOS_SIGNING_IDENTITY <string>
  [MACOS_SIGNING_KEYCHAIN <string>]
  [MACOS_ENTITLEMENTS <entitlement...>]
  [WINDOWS_ICON <path>]
  WINDOWS_SIGNING_SUBJECT <string>
  WINDOWS_SIGNING_THUMBPRINT <string>
  [LINUX_ICON <path>]
  LINUX_CATEGORY <string>
)
```
```

File: README.md [hunk 4]
```diff
@@ -1,8 +1,110 @@
...
+Thumbprint of the Windows code signing certificate.\
+Default  :
+
+##### `LINUX_ICON <path>` 
+Path to the icon for the Linux AppImage.\
+Default  :
+
+##### `LINUX_CATEGORY <string>` 
+The category for the app in Linux application menus.\
+Default  :
```
Comment (kasperisager) [inline README.md]:
```suggestion

##### `<target>`

The name of the CMake target to create for the Pear appling.

##### `KEY <string>`

The key of the Pear appling.

##### `NAME <string>`

The name of the appling as presented to users.

##### `VERSION <string>` 

The semantic version of the Pear appling, e.g. "1.0.0".

##### `DESCRIPTION <string>`

A short description of the app's functionality.

##### `AUTHOR <string>`

Author's name or the name of the organization creating the appling.

##### `SPLASH <path>`

The path to a splash screen image displayed during appling launch. Defaults to `"assets/splash.png"`.

##### `MACOS_ICON <path>` 

The path to the icon for the macOS app bundle. Defaults to `"assets/darwin/icon.png"`.

##### `MACOS_CATEGORY <string>`

The category for the app in the macOS App Store or Finder.

##### `MACOS_IDENTIFIER <string>`

A unique bundle identifier for the macOS app.

##### `MACOS_SIGNING_IDENTITY <string>`

A macOS code signing identity string. Defaults to `"Apple Development"`.

##### `MACOS_SIGNING_KEYCHAIN <string>` 

The path to the keychain containing the signing identity.

##### `MACOS_ENTITLEMENTS <entitlement...>` 

A list of macOS entitlements for special permissions.

##### `WINDOWS_ICON <path>` 

The path to the icon for the Windows MSIX package. Defaults to `"assets/win32/icon.png"`.

##### `WINDOWS_SIGNING_SUBJECT <string>` 

Subject name for Windows code signing.

##### `WINDOWS_SIGNING_THUMBPRINT <string>` 

Thumbprint of the Windows code signing certificate.

##### `LINUX_ICON <path>` 

Path to the icon for the Linux AppImage. Defaults to `"assets/linux/icon.png"`.

##### `LINUX_CATEGORY <string>` 

The category for the app in Linux application menus.
```

**Review by kasperisager — CHANGES_REQUESTED** [review]
Looks good! I've made some editorial and formatting changes and filled in the defaults.

**Comment by heyitsvi** [comment]
changes made as requested, let me know

**Review by kasperisager — APPROVED** [review]

**Pull request merged**